docs/asynchronous-processing/product-recalculations.md

```diff
@@ -115,11 +115,56 @@ That way we are sure the code works – the message is truly dispatched, thus th
     Calling `handleDispatchedRecalculationMessages()` method creates a snapshot in Elasticsearch before any changes are exported and restores it afterward.<br>
     Tests are not dependent on the changes made or the order of run.
 
+## Recalculation scope
+
+In many occasions, to optimize the application, you might need to restrict the scope of a product recalculation and elastic export.
+Sometimes you do not need to recalculate visibility or selling denied, or you need to export just particular fields into Elasticsearch.
+For example, when launching a new domain, you need to export just the product URLs into Elasticsearch which takes significantly less time than the full recalculation and export.
+On the other hand, it might be hard to keep in mind all the dependencies (e.g. you need to know you need to recalculate visibility after you change `Product::$sellingFrom`,
+or you need to keep in mind that with a change of product URL, you always need to export `hreflang_links` into Elasticsearch as well, etc.).
+
+For this purpose, recalculation scopes are defined as an associative array in the `Shopsys\FrameworkBundle\Model\Product\Elasticsearch\Scope\ProductExportScopeConfigFacade` class.
+The scopes are represented by instances of the `Shopsys\FrameworkBundle\Model\Product\Elasticsearch\Scope\ProductExportScopeRule` class and indexed by their names.
+Each scope rule defines which fields should be exported to Elasticsearch together (`$productExportFields` property) and whether some actions (recalculations) should be done before the export (`$productExportPreconditions` property).
+
+You can use `./bin/console shopsys:list:export-scopes` command to list and examine all the available scopes.
```

Comment:
Consider adding this and other new commands to `check-console-commands` and proper part in the build of a fork in `docker-build.yaml`

Reply:
good point, added in [`273ab33` (#3051)](https://github.com/shopsys/shopsys/pull/3051/commits/273ab3328c5b899cb49621747892c5a1c2f512df)